### PR TITLE
Update adventure.py

### PR DESCRIPTION
**Pull request by Ririsama**

### CODE REVIEW

File: adventure/adventure.py [hunk 1]
```diff
@@ -2936,15 +2938,32 @@ def E(t: str) -> str:
...
+                continue
+            bonus_cleric = int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3)
+            if c.heroclass["name"] == "Cleric" and not aura:
+                aura_chance = min(int(bonus_cleric / 2 + 1), c.lvl)
+                aura_roll = random.randint(1, 100)
+                if aura_roll in range (1, aura_chance):
+                    aura = True
+                    aura_bonus = int(aura_chance * 0.2)
+                    msg += (
+                        f"An holy aura starts surrounding {bold(self.E(user.display_name))} while praying! "
```
Comment (oddslol) [inline adventure/adventure.py]:
```suggestion
                        f"A holy aura starts surrounding {bold(self.E(user.display_name))} while praying! "
```
Confusingly... "an" can be used before some words that start with "h" like "an hour" but normally when the "h" sound isn't really pronounced... In the case of "holy" you pronounce the "h" so it's best to say "a holy" not "an holy"

File: adventure/adventure.py [hunk 2]
```diff
@@ -3068,71 +3087,72 @@ def E(t: str) -> str:
...
-                    msg += (
-                        f"{bold(self.E(user.display_name))} blessed like a "
-                        "madman but nobody was there to receive it.\n"
-                    )
-                elif c.heroclass["name"] == "Cleric" and c.heroclass["ability"]: #small bonus if the user is cleric with activated ability
+            pray_score = pray_bonus + roll
+            if total_size == 0 or roll == 1: #fumble in case no one is fighting, talking, using magic or if roll 1
+                if total_size == 0:
+                    msg += f"{bold(humanize_list(pray_list_name))} blessed like "
+                    if len(pray_list) == 1:
```
Comment (oddslol) [inline adventure/adventure.py]: could be simplified to something like:
`attrib = "a madman" if len(pray_list) == 1 else "madmen"`
then use attrib inside a single f string

File: adventure/adventure.py [hunk 3]
```diff
@@ -3068,71 +3087,72 @@ def E(t: str) -> str:
...
+        for user in magic_list:
             try:
                 c = await Character._from_json(self.config, user)
             except Exception:
                 log.error("Error with the new character sheet", exc_info=True)
                 continue
-            bonus_cleric = 0                   
-            if c.heroclass["name"] == "Cleric":
-                bonus_cleric = int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3)
+            if c.heroclass["name"] == "Wizard" and not glyphs and len(pray_list) >= 1:
```
Comment (oddslol) [inline adventure/adventure.py]: Is it better to check "len(pray_list) >= 1" outside of the magic_list loop? 
Probably best to check before entering the loop... to save us iterating through all of the users if we don't have to

File: adventure/adventure.py [hunk 4]
```diff
@@ -3068,71 +3087,72 @@ def E(t: str) -> str:
...
+                if total_size == 0:
+                    msg += f"{bold(humanize_list(pray_list_name))} blessed like "
+                    if len(pray_list) == 1:
+                        msg += f"a madman "
+                    else:
+                        msg += f"madmen "
+                    msg += f"but nobody was there to receive it.\n"
+                elif c.heroclass["name"] == "Cleric" and c.heroclass["ability"]: #malus but no fumble if the user is cleric with activated ability
+                    contrib_attack = int(((len(fight_list) / total_size) * pray_score - len(fight_list)) * (1 + (glyphs_bonus / 100)))
+                    contrib_diplomacy = int((len(talk_list) / total_size) * pray_score - len(talk_list)) * (1 + (glyphs_bonus / 100)))
```
Comment (oddslol) [inline adventure/adventure.py]:
```suggestion
                    contrib_diplomacy = int(((len(talk_list) / total_size) * pray_score - len(talk_list)) * (1 + (glyphs_bonus / 100)))
```
missed a bracket

File: adventure/adventure.py (hunk 3 above)
Comment (Ririsama) [reply adventure/adventure.py]: Fair point

File: adventure/adventure.py (hunk 2 above)
Comment (Ririsama) [reply adventure/adventure.py]: Sure why not

Comment (Ririsama) [reply adventure/adventure.py]: done

File: adventure/adventure.py (hunk 3 above)
Comment (Ririsama) [reply adventure/adventure.py]: done

**Pull request merged**